includes/html/pages/routing/bgp.inc.php

```diff
@@ -272,22 +272,26 @@
         // display overlib graphs
         $graph_array = [];
         $graph_array['type'] = 'bgp_updates';
-        $graph_array['id'] = $peer['bgpPeer_id'];
         $graph_array['to'] = \LibreNMS\Config::get('time.now');
         $graph_array['from'] = \LibreNMS\Config::get('time.day');
         $graph_array['height'] = '110';
         $graph_array['width'] = $width;
 
         // Peer Address
-        $graph_array_zoom = $graph_array;
-        $graph_array_zoom['height'] = '150';
-        $graph_array_zoom['width'] = '500';
-        $overlib_link = 'device/device=' . $peer['device_id'] . '/tab=routing/proto=bgp/';
-        $peeraddresslink = '<span class=list-large>' . \LibreNMS\Util\Url::overlibLink($overlib_link, $peer_addr, \LibreNMS\Util\Url::graphTag($graph_array_zoom)) . '</span>';
+        $peer_device = \App\Models\Device::whereHas('bgppeers', fn ($q) => $q->where('bgpLocalAddr', $peer['bgpPeerIdentifier']))->first();
+        if (! empty($peer_device)) {
+            $peeraddresslink = '<span class=list-large>' . \LibreNMS\Util\Url::deviceLink($peer_device, $peer_addr, vars: ['tab' => 'routing', 'proto' => 'bgp']) . '</span>';
```

Comment:
Hmm, looking at deviceLink, I think it should return the second argument if the first is not a device.
Also, you don't need the name (vars:) for the third argument here.  I only put it in last time so I could omit the second argument.

Reply:
Fixed, hopefully I didn't break it.

Reply:
Nice fix/addition :)!